statusHandler/statusMetricsProvider.go

```diff
@@ -97,40 +110,62 @@ func (sm *statusMetrics) Close() {
 
 // StatusMetricsMapWithoutP2P will return the non-p2p metrics in a map
 func (sm *statusMetrics) StatusMetricsMapWithoutP2P() map[string]interface{} {
-	statusMetricsMap := make(map[string]interface{})
-	sm.nodeMetrics.Range(func(key, value interface{}) bool {
-		keyString := key.(string)
-		if strings.Contains(keyString, "_p2p_") {
-			return true
-		}
-
-		statusMetricsMap[key.(string)] = value
-		return true
+	return sm.getMetricsWithKeyFilterMutexProtected(func(input string) bool {
+		return !strings.Contains(input, "_p2p_")
 	})
-
-	return statusMetricsMap
 }
 
 // StatusP2pMetricsMap will return the p2p metrics in a map
 func (sm *statusMetrics) StatusP2pMetricsMap() map[string]interface{} {
+	return sm.getMetricsWithKeyFilterMutexProtected(func(input string) bool {
+		return strings.Contains(input, "_p2p_")
+	})
+}
+
+func (sm *statusMetrics) getMetricsWithKeyFilterMutexProtected(filterFunc func(input string) bool) map[string]interface{} {
 	statusMetricsMap := make(map[string]interface{})
-	sm.nodeMetrics.Range(func(key, value interface{}) bool {
-		keyString := key.(string)
-		if !strings.Contains(keyString, "_p2p_") {
-			return true
+
+	sm.mutUint64Operations.RLock()
+	for key, value := range sm.uint64Metrics {
+		if !filterFunc(key) {
+			continue
 		}
 
-		statusMetricsMap[key.(string)] = value
-		return true
-	})
+		statusMetricsMap[key] = value
+	}
+	sm.mutUint64Operations.RUnlock()
+
+	sm.mutStringOperations.RLock()
+	for key, value := range sm.stringMetrics {
+		if !filterFunc(key) {
+			continue
+		}
+
+		statusMetricsMap[key] = value
```

Comment:
if the same key is present in multiple maps, the previous key/value will be overwritten. There are no cases with same keys in different maps, right?